File: src/com/Torvald/Terrarum/Actors/Faction/FactionRelatorFactory.java
package com.Torvald.Terrarum.Actors.Faction;

import com.Torvald.JsonFetcher;
import com.google.gson.JsonObject;

import java.io.IOException;

/**
 * Created by minjaesong on 16-02-15.
 */
public class FactionRelatorFactory {

    private static final String JSONPATH = "./res/raw/";

    public Faction build(String filename) throws IOException {
        JsonObject jsonObj = JsonFetcher.readJson(JSONPATH + filename);
        Faction factionObj = new Faction(jsonObj.get("factionname").getAsString());


        jsonObj.get("factionamicable").getAsJsonArray().forEach(
                s -> factionObj.addFactionAmicable(s.getAsString())
        );
        jsonObj.get("factionneutral").getAsJsonArray().forEach(
                s -> factionObj.addFactionNeutral(s.getAsString())
        );
        jsonObj.get("factionhostile").getAsJsonArray().forEach(
                s -> factionObj.addFactionHostile(s.getAsString())
        );
        jsonObj.get("factionfearful").getAsJsonArray().forEach(
                s -> factionObj.addFactionFearful(s.getAsString())
        );

        return factionObj;
    }

}

File: src/com/Torvald/Terrarum/Actors/Faction/FactionRelatorFactory.kt
package com.Torvald.Terrarum.Actors.Faction

import com.Torvald.JsonFetcher
import com.google.gson.JsonObject

import java.io.IOException

/**
 * Created by minjaesong on 16-02-15.
 */
class FactionRelatorFactory {

    @Throws(IOException::class)
    fun build(filename: String): Faction {
        val jsonObj = JsonFetcher.readJson(JSONPATH + filename)
        val factionObj = Faction(jsonObj.get("factionname").asString)


        jsonObj.get("factionamicable").asJsonArray.forEach { s -> factionObj.addFactionAmicable(s.asString) }
        jsonObj.get("factionneutral").asJsonArray.forEach { s -> factionObj.addFactionNeutral(s.asString) }
        jsonObj.get("factionhostile").asJsonArray.forEach { s -> factionObj.addFactionHostile(s.asString) }
        jsonObj.get("factionfearful").asJsonArray.forEach { s -> factionObj.addFactionFearful(s.asString) }

        return factionObj
    }

    companion object {

        val JSONPATH = "./res/raw/"
    }

}
